lib/oxidized/model/tplink.rb

```diff
@@ -34,27 +39,39 @@ class TPLink < Oxidized::Model
     cfg
   end
 
-  cmd 'show system-info' do |cfg|
+  cmd "show system-info" do |cfg|
     comment cfg.each_line.to_a[3..-3].join
   end
 
-  cmd 'show running-config' do |cfg|
+  cmd "show running-config" do |cfg|
     lines = cfg.each_line.to_a[1..-1]
     # cut config after "end"
     lines[0..lines.index("end\n")].join
   end
 
   cfg :telnet, :ssh do
-    username /^User ?[nN]ame:/
-    password /^\r?Password:/
+    username /^(User ?[nN]ame|User):/
+    password /^\r?[pP]assword:/
   end
 
   cfg :telnet, :ssh do
-    if vars :enable
-      post_login do
-        send "enable\r"
-        cmd vars(:enable)
+    post_login do
+      if vars(:enable) == true
+        cmd "enable"
+      elsif vars(:enable)
+        cmd "enable", /^[pP]assword:/
+        send vars(:enable) + "\n\r"
       end
+      # disable paging
+      cmd "terminal length 0"
```

Comment:
T1700G-28TQRev2 v 2.0.1 (supported, but has no newer fw) doesn't have the command, and it has paging enabled. i'd recommend not to remove the old handling, yet).

Reply:
Oh, I thought that it maybe should be useful on older firmwares and already reverted the changes in 5470318eac27d94635fd40fe3a0be60174eed204

Reply:
Maybe handy to know, T1700G-28TQRev2 can use the V3 firmware. I have it running for over a year. 
Support told me to upgrade because of some issues i had which where fixed in the V3 firmware.
But also `terminal length 0` does not exist in the v3 firmware, sadly

Reply:
Well, now both modes are supported, the code to handle paging is still there and, if supported, the `terminal length 0` will works.

Reply:
Older devices does not have "terminal length 0". Maybe we should add a new model? Or make it configurable to use newer commands?

Reply:
I don't understand the problem with the code as-is. It will send the `terminal length 0`, if it works, paging will be disabled. If not works Oxidized will get a `Error: Bad command` and the backup process will continue. If the `Press any key to continue` prompt appears, Oxidized will send `<space>` and the output will scroll until the end.

This will work with devices that have the `terminal length 0` and with those that doesn't have it.

If this is a problem it's easier to remove the `terminal length 0` command. It isn't an impeditive to the backup process.